### PR TITLE
isograph sampling

### PR DESCRIPTION
**Pull request by chu-andrew**
## Description
- Sets up technical challenge 2
- Adds ability to sample points a given distance away from an origin in radial directions
- Modifies fetchRouteMatrix to accept coordinate points as origins/destinations, instead of only addresses, which is necessary for the proposed coordinate-based sampling system
- Displays samples on map with pins--- this will be used for development and testing of the isograph generation before the visualization is developed

## Views
Samples shown on map, with the number of directions = 8 and number of samples per direction = 2.
image

### CODE REVIEW

File: backend/IsographUtils.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+async function fetchCosts(origin, samplePoints, costType, departureTime) {
+  const routesData = await fetchResultsUtils.fetchRouteMatrix(
+    origin,
+    samplePoints,
+    departureTime,
+    true
+  );
+
+  if (costType === "duration") {
```
Comment (min2033) [inline backend/IsographUtils.js:30]: nit: can we use consts or enums for these?

File: backend/IsographUtils.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    parseCostFunction = fetchResultsUtils.parseDuration;
+  } else if (costType === "fare") {
+    parseCostFunction = fetchResultsUtils.calculateFare;
+  } else {
+    console.error("Cost type not supported.");
```
Comment (min2033) [inline backend/IsographUtils.js:35]: nit: use const

File: backend/IsographUtils.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+function findCoordinate(origin, distance, direction) {
+  const EARTH_RADIUS = 6371000; // meters
+
+  const deltaMetersNorth = Math.abs(distance) * Math.cos(toRadians(direction));
+  const deltaMetersEast = Math.abs(distance) * Math.sin(toRadians(direction));
+
+  const [latitude, longitude] = origin;
+  return [
+    latitude + (deltaMetersNorth / EARTH_RADIUS) * (180 / Math.PI),
```
Comment (min2033) [inline backend/IsographUtils.js:11]: nit: pull (180 / Math.PI) as const

File: backend/IsographUtils.js (hunk 1 above)
Comment (chu-andrew) [reply backend/IsographUtils.js:30]: Addressed in 355b6af95fc6ee4ca65267c2818f209a1ad10074.

File: backend/IsographUtils.js (hunk 2 above)
Comment (chu-andrew) [reply backend/IsographUtils.js:35]: Addressed in 355b6af95fc6ee4ca65267c2818f209a1ad10074.

File: backend/IsographUtils.js (hunk 3 above)
Comment (chu-andrew) [reply backend/IsographUtils.js:11]: Addressed in 355b6af95fc6ee4ca65267c2818f209a1ad10074.

**Pull request merged**